src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,43 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  userId: number,
+  todoId: number,
+  onSelectTodo: (id: number) => void,
+  onSelectUser: (userId: number) => void,
+};
+
+export const TodoModal: React.FC<Props> = ({
+  todos,
+  userId,
+  todoId,
+  onSelectTodo,
+  onSelectUser,
+}) => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      const userData = await getUser(userId);
+
+      setSelectedUser(userData);
+    };
+
+    fetchUser();
+  }, []);
```

Comment:
Add `[userId]` as a dependency to the `useEffect` to refetch the user data when `userId` changes.